language/vm/vm_runtime/vm_runtime_types/src/native_functions/dispatch.rs

```diff
@@ -197,6 +197,21 @@ lazy_static! {
             vec![MutableReference(Box::new(tstruct(addr, "Vector", "T", vec![TypeParameter(0)])))],
             vec![TypeParameter(0)]
         );
+        add!(m, addr, "Vector", "reverse",
```

Comment:
Do we want to implement these natively? Both can be implemented in pure Move, and doing it that way will save us from specifying gas costs for two new native operations + the need to specify the semantics for the formalism and verifier.

Reply:
That is exactly the same question I have in mind. It seems to me that reverse can be done by poping out the input vector and pushing to a new vector. It's just that it would be slower.

Reply:
Agreed on implementing reverse this way.

We could implement append this way similarly (reverse other vector, and then do a reverse with a starting vector of lhs). But, this would be more costly.

The more I'm thinking about it, the more I think it's better that these are not native. I'll fix and update the diff in a bit.

Reply:
I think starting with a pure Move implementation when possible, then considering a switch to `native` if performance/gas cost becomes a problem is a reasonable strategy. When/if we do this, we can even use the pure Move implementation as a spec for native implementers :).